Amazon.Ion.ObjectMapper.Test/TestObjects.cs

```diff
@@ -49,17 +49,68 @@ public override string ToString()
     [IonDoNotAnnotateType(ExcludeDescendants = true)]
     public class Motorcycle : Vehicle
     {
-        public string Brand { get; init; }
+        public string Make { get; init; }
+
+        [IonField]
+        public string color;
+
+        [IonField]
+        public bool canOffroad;
+        
+        public override string ToString()
+        {
+            return $"<Motorcycle>{{ Make: {this.Make}, color: {this.color}, canOffroad: {this.canOffroad} }}";
+        }
+    }
+
+    // For testing case insensitive deserialization.
+    public class MOTORCYCLE : Vehicle
+    {
+        public string MAKE { get; init; }
+
+        [IonField]
+        public string COLOR;
+
+        [IonField]
+        public bool CANOFFROAD;
+        
+        public override string ToString()
+        {
+            return $"<MOTORCYCLE>{{ MAKE: {this.MAKE}, COLOR: {this.COLOR}, CANOFFROAD: {this.CANOFFROAD} }}";
+        }
+    }
+    
+    // For testing case insensitive deserialization.
+    public class motorcycle : Vehicle
+    {
+        public string make { get; init; }
 
         [IonField]
         public string color;
 
         [IonField]
-        public bool canOffroad; 
+        public bool canoffroad;
```

Comment:
This is a field. Fields are not affected by this option.

Reply:
Correct. And that is reflected in the test.

The original input has a non-null `color` field and a true `canOffroad` field and the test verifies that the difference in case results in those fields not being serialized/deserialized:
```c#
Assert.IsNull(deserialized1.COLOR);
Assert.IsFalse(deserialized1.CANOFFROAD);
```